File: AdBoard.Api.Spring/src/main/kotlin/ru/atom/adboard/dal/entities/Subcategory.kt
package ru.atom.adboard.dal.entities

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "subcategories")
data class Subcategory(
    @Id
    @Column(name = "id", columnDefinition = "UUID")
    val id: UUID = UUID.randomUUID(),

    @Column(name = "name", nullable = false)
    val name: String,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    @JsonIgnoreProperties("subcategories")
    val category: Category,

    @OneToMany(mappedBy = "subcategory", fetch = FetchType.LAZY)
    @JsonIgnore
    val announcements: MutableSet<Announcement> = mutableSetOf()
)

File: AdBoard.API/AdBoard.DAL/Entities/Subcategory.cs
using System;
using System.Collections.Generic;

namespace AdBoard.DAL.Entities;

public class Subcategory
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public Guid CategoryId { get; set; }

    public ICollection<Announcement> Announcements { get; set; } = new List<Announcement>();
    public Category Category { get; set; } = null!;
}
